packages/SwingSet/src/kernel/vatAdmin/vatAdmin-src.js

```diff
@@ -0,0 +1,80 @@
+import harden from '@agoric/harden';
+
+/**
+ * A Vat management device that provides a capability that can be used to
+ * create new vats. create(adminId, code) creates a new vat running code, and
+ * returns the root object.
+ *
+ * setup(...) calls makeDeviceSlots(..., makeRootDevice, ...), which calls
+ * deviceSlots' build() function (which invokes makeRootDevice) to create the
+ * root device. Selected vats that need to create new vats can be given access
+ * to the device.
+ *
+ * This code runs in the inner part of the device vat. It handles kernel objects
+ * in serialized format, and can use SO() to send messages to them. The only
+ * device object exposed to vats is the vat creator. The root objects for the
+ * new vats are returned as javascript objects.
+ *
+ * We don't currently need to use SO, or to maintain state. All functionality is
+ * provided by calling kernel functions and passing in the VatID. The wrapper
+ * vat sends the VatID on every call.
+ */
+export default function setup(syscall, state, helpers, endowments) {
```

Comment:
if you change the endowments as described above, then at the start of this function we could do `let { create } = endowments;`

Reply:
done